### PR TITLE
Added authentication portal admin advisory banner

### PR DESCRIPTION
**Pull request by mpmadhavig**
### Related Issues
- https://github.com/wso2/product-is/issues/15521

### Related PRs
- https://github.com/wso2-extensions/identity-governance/pull/667
- https://github.com/wso2/carbon-identity-framework/pull/4480
- https://github.com/wso2/product-is/pull/15618

### Feature
Fig 1: Admin login page with the advisory banner
Screenshot 2023-03-14 at 12 34 41

Fig 2: Management console: Admin advisory banner configuration
Screenshot 2023-03-14 at 12 35 22

Fig 3: Console Application: Admin advisory banner configuration
Screenshot 2023-03-14 at 12 35 06

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [x] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Comment by pavinduLakshan** [comment]
Shall we add a screenshot of how this banner would look like, either to the issue or this PR?

File: apps/authentication-portal/src/main/webapp/basicauth.jsp
Comment (pavinduLakshan) [inline apps/authentication-portal/src/main/webapp/basicauth.jsp:296]: Shall we use `data-componentid` instead? we are moving away from `data-testid` therefore new code should not have `data-testid`.

Comment (pavinduLakshan) [inline apps/authentication-portal/src/main/webapp/basicauth.jsp:296]: Shall we use `data-componentid` instead? we are moving away from `data-testid` therefore new code should not have `data-testid`.

```suggestion
    <div class="ui warning message" data-componentid="login-page-admin-session-advisory-banner">
```

**Comment by mpmadhavig** [comment]
> Shall we add a screenshot of how this banner would look like, either to the issue or this PR?

Added.

File: modules/i18n/src/models/namespaces/console-ns.ts
Comment (pavinduLakshan) [inline modules/i18n/src/models/namespaces/console-ns.ts:3736]: What is the usage of these strings? couldn't see any usage in this PR though. Maybe you are planning to send the configuration UI impl in a follow-up PR?

File: apps/authentication-portal/src/main/webapp/basicauth.jsp
Comment (pavinduLakshan) [inline apps/authentication-portal/src/main/webapp/basicauth.jsp:297]: Just asking out of curiosity, will this content be i18n compatible?

**Comment by pavinduLakshan** [comment]
> > Shall we add a screenshot of how this banner would look like, either to the issue or this PR?
> 
> Added.

Thanks. Is the banner content finalised already? It'd better to initiate a discussion around this new addition, as most of the team may not be aware of this IMO. just a suggestion, apologies if this has already been discussed.

File: apps/authentication-portal/src/main/webapp/basicauth.jsp
Comment (pavinduLakshan) [inline apps/authentication-portal/src/main/webapp/basicauth.jsp:295]:
```suggestion
<% if (isAdminSessionAdvisoryControlEnabledInTenant) { %>
```

File: modules/i18n/src/models/namespaces/console-ns.ts
Comment (mpmadhavig) [reply modules/i18n/src/models/namespaces/console-ns.ts:3736]: According to the governance connector implementation, you only need to define the newly created connector under the relavant connectorCategory. Then the connectors defined under connectorCategories are being automatically displayed in the relavant pages. 

Please refer [const resolveConnectorTitle = (connector: GovernanceConnectorInterface): ReactNode => {](https://github.com/wso2/identity-apps/blob/74b275743a60eb7032843f23bf55f157ec35aebe/apps/console/src/features/server-configurations/components/governance-connectors/dynamic-governance-connector.tsx#L207) for more details.

**Comment by mpmadhavig** [comment]
> Thanks. Is the banner content finalised already? IMO, It'd be better to initiate a discussion around this new addition, as most of the team may not be aware of this. just a suggestion, apologies if this has already been discussed.

The content is not finalized yet. But since this feature is coming as a request from the security team, we decided to take an input from them and get approval by the docs team.

File: apps/authentication-portal/src/main/webapp/basicauth.jsp
Comment (mpmadhavig) [reply apps/authentication-portal/src/main/webapp/basicauth.jsp:297]: Thank you for bringing this up. According my knowledge, it is not. I will update the PR with the necessary changes.

**Review by pavinduLakshan — DISMISSED** [review]

File: apps/authentication-portal/src/main/webapp/basicauth.jsp
Comment (mpmadhavig) [reply apps/authentication-portal/src/main/webapp/basicauth.jsp:297]: Since this is a user editable field, there is no way for us to make it i18n compatible.

Comment (mpmadhavig) [reply apps/authentication-portal/src/main/webapp/basicauth.jsp:297]: Created the issue https://github.com/wso2/product-is/issues/15630 to address this in a separate effort.

Comment (pavinduLakshan) [reply apps/authentication-portal/src/main/webapp/basicauth.jsp]: Thanks for taking time to check this.

> Created the issue [wso2/product-is#15630](https://github.com/wso2/product-is/issues/15630) to address this in a separate effort.

+1

**Review by pavinduLakshan — DISMISSED** [review]

File: apps/authentication-portal/src/main/webapp/basicauth.jsp
Comment (rajithacharith) [inline apps/authentication-portal/src/main/webapp/basicauth.jsp:297]: Shouldn't this encode (if `adminAdvisoryBannerContentOfTenant` is an user input)
```suggestion
        <%=Encode.forHtml(adminAdvisoryBannerContentOfTenant)%>
```

**Comment by mpmadhavig** [comment]
The implementation path has been changed and therefore the feature will be sent with the following effort.

1. https://github.com/wso2/carbon-identity-framework/pull/4512
2. https://github.com/wso2/carbon-kernel/pull/3560
3. https://github.com/wso2/identity-api-server/pull/441
4. https://github.com/wso2/identity-apps/pull/3872
5. https://github.com/wso2/product-is/pull/15700

**Pull request closed without merging**
